staging/src/k8s.io/mount-utils/mount_linux.go

```diff
@@ -667,3 +674,82 @@ func forceUmount(path string) error {
 	}
 	return nil
 }
+
+// isMountFastPath is a method of detecting a mount that really fast.
+// It should return a true and no-error ONLY when it is an actual mount point.
+// In cases, where it is not guaranteed or cannot be determined, it should return an error.
+// For Linux, it avoids parsing /proc/mounts which is a very slow operation.
+func isMountFastPath(path string) (bool, error) {
+	// the second parameter should be nil always in production code path.
+	// it is used to inject mock function for openat2 for unit-testing.
+	return isMountPointUsingOpenAt2(path, nil)
+}
+
+// isMountPointUsingOpenAt2 uses openat2 syscall is a method of detecting a mount that works for all kinds
+// of mounts (incl. bind mounts), but requires a recent (v5.6+) linux kernel.
+// The code takes inspiration from here:
+// https://github.com/moby/sys/blob/40883be4345cf291513e2a4112896d393455fdfb/mountinfo/mounted_linux.go#L10
+// It expects that all symlinks have been evaluated.
+// If an ENOENT is returned by openat2, return os.ErrNotExist to satisfy isNotMountPoint return values.
+// openat2 is injected for testing purposes. In production code path, it should always be nil.
```

Comment:
Errors from `isMountFastPath` are always only compared to `nil` and then thrown away. They're not returned from `isNotMountPoint()`.